XenModel/Actions/VMPP/ChangePolicyEnabledAction.cs

```diff
@@ -39,24 +39,39 @@
 
 namespace XenAdmin.Actions
 {
-    public class ChangePolicyEnabledAction : PureAsyncAction
+    public class ChangePolicyEnabledAction<T> : PureAsyncAction where T : XenObject<T>
     {
-        private VMPP _vmpp;
-        public ChangePolicyEnabledAction(VMPP vmpp)
-            : base(vmpp.Connection, string.Format(Messages.CHANGE_POLICY_STATUS, vmpp.Name))
+        private IVMPolicy _policy;
+        public ChangePolicyEnabledAction(IVMPolicy policy)
+            : base(policy.Connection, string.Format(Messages.CHANGE_POLICY_STATUS, policy.Name))
         {
-            _vmpp = vmpp;
-            Pool = Helpers.GetPool(vmpp.Connection);
+            _policy = policy;
+            Pool = Helpers.GetPool(_policy.Connection);
         }
 
         protected override void Run()
         {
-            bool value = !_vmpp.is_policy_enabled;
-            Description = value ? string.Format(Messages.ENABLING_VMPP, _vmpp.Name) : string.Format(Messages.DISABLING_VMPP, _vmpp.Name);
-            
-            VMPP.set_is_policy_enabled(Session, _vmpp.opaque_ref, !_vmpp.is_policy_enabled);
-            Description = value ? string.Format(Messages.ENABLED_VMPP, _vmpp.Name) : string.Format(Messages.DISABLED_VMPP, _vmpp.Name);
-            
+            bool value = !_policy.is_enabled;
+            if (typeof(T) == typeof(VMPP))
+            {
+                Description = value ? string.Format(Messages.ENABLING_VMPP, _policy.Name) : string.Format(Messages.DISABLING_VMPP, _policy.Name);

```

Comment:
Again in this area, let's move the typeof(T) == typeof(VMPP) inside the string.Format.